nomad/resource_acl_policy.go

```diff
@@ -43,10 +44,78 @@ func resourceACLPolicy() *schema.Resource {
 				Required:    true,
 				Type:        schema.TypeString,
 			},
+
+			"workload_identity": {
+				Description: "Workload identity that this should be applied to.",
+				Optional:    true,
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"namespace": {
+							Description: "Namespace",
+							Type:        schema.TypeString,
+							Optional:    true,
+							Default:     "default",
+						},
+						"job_id": {
+							Description: "Job",
+							Type:        schema.TypeString,
+							Required:    true,
+						},
+						"group": {
+							Description: "Group",
+							Type:        schema.TypeString,
+							Optional:    true,
+						},
+						"task": {
+							Description: "Task",
+							Type:        schema.TypeString,
+							Optional:    true,
+						},
+					},
+				},
+			},
```

Comment:
```suggestion
			},
			ValidateFunc: func(i interface{}, k string) ([]string, []error) {
				// ...
			},
```

There are some validations rules applied here that are static enough that I think would be worth checking at the provider so they are caught during plan:
https://github.com/hashicorp/nomad/blob/c7c8bd4ce2fcd9f2cd13278043b7baf1f8dd4f47/nomad/structs/structs.go#L12373-L12386

Reply:
I made this change but I get an error `ValidateFunc is not yet supported on lists or sets`. It's also unclear if there are plans or willingness to add support for this either: https://github.com/hashicorp/terraform-plugin-sdk/issues/156

Reply:
I'm instead moving all the validation to the parse function where I had some partial validation before.